backend/src/db.js

```diff
@@ -0,0 +1,78 @@
+import { DataSource } from 'apollo-datasource'
+import { Post } from './models/Post'
+import { User } from './models/User'
+
+export class InMemoryNewsDS extends DataSource {
+  constructor () {
+    super()
+    this.nextPostId = 0
+    this.posts = new Map()
+    this.users = new Map()
+  }
+
+  createUser (name) {
+    if (this.users.has(name)) throw new Error(`User with name ${name} already exists!`)
+    this.users.set(name, new User(name))
+  }
+
+  createPost (title, votes, authorName) {
+    const author = this.users.get(authorName)
+    if (author === undefined) throw new Error(`User with name ${authorName} does not exist!`)
+    const post = new Post(String(this.nextPostId++), title, votes, author)
+    this.posts.set(post.id, post)
+    this.users.get(authorName).posts.add(post)
+    return post
+  }
+
+  deletePost (id) {
+    const post = this.posts.get(id)
+    if (post === undefined) throw new Error(`No post found for ID ${id}!`)
+    this.posts.delete(id)
+    this.users.get(post.author.name).posts.delete(post)
+    return post
+  }
```

Comment:
You could consider using type resolvers to add all posts that are written by a certain user. Then you don't have to add and delete posts from users directly and you could just save the authors name in the post object and not worry about it in your db.